crates/nu-command/src/input_handler.rs

```diff
@@ -49,7 +49,16 @@ where
     C: Fn(&Value, &A, Span) -> Value + Send + Sync + 'static + Clone + Copy,
 {
     match arg.take_cell_paths() {
-        None => input.map(move |v| cmd(&v, &arg, v.span().unwrap_or(span)), ctrlc),
+        None => input.map(
+            move |v| {
+                match v {
+                    // Propagate errors inside the input
+                    Value::Error { .. } => v,
+                    _ => cmd(&v, &arg, span),
+                }
+            },
+            ctrlc,
+        ),
```

Comment:
I don't have any reason to believe that this *will* cause problems, but it's an interesting change that affects a lot of commands. Something to take a look at if anything breaks unexpectedly.